io.openems.edge.timedata.influxdb/src/io/openems/edge/timedata/influxdb/TimedataInfluxDbImpl.java

```diff
@@ -216,17 +216,39 @@ public SortedMap<Long, SortedMap<ChannelAddress, JsonElement>> queryResendData(Z
 		return emptySortedMap();
 	}
 
+
+	
 	@Override
 	public CompletableFuture<Optional<Object>> getLatestValue(ChannelAddress channelAddress) {
-		// TODO implement this method
-		return CompletableFuture.completedFuture(Optional.empty());
+	    return CompletableFuture.supplyAsync(() -> {
+	        try {
+	            SortedMap<ZonedDateTime, SortedMap<ChannelAddress, JsonElement>> result = this.influxConnector.queryLastData(
+	                Optional.empty(), channelAddress, this.config.measurement());
+
+	            // Retrieve the latest value from the result
+	            if (!result.isEmpty()) {
+	                SortedMap<ChannelAddress, JsonElement> latestValues = result.get(result.lastKey());
+	                if (latestValues.containsKey(channelAddress)) {
+	                    JsonElement element = latestValues.get(channelAddress);
+	                    if (element.isJsonPrimitive() && element.getAsJsonPrimitive().isNumber()) {
+	                    	return Optional.of(element.getAsLong());
```

Comment:
what i usually prefer to avoid nested if statements in general is to just return inside ifs so the first if here wouldnt be 
`if (!result.isEmpty()) { ... }` it would be like `if (result.isEmpty()) { return Optional.empty(); } ...`

but thats up to you 3 nestings isnt that bad.

[More detailed description](https://stackoverflow.com/a/47619546)